php_phongo.c

```diff
@@ -395,38 +395,43 @@ zend_bool phongo_writeconcernerror_init(zval *return_value, bson_t *bson TSRMLS_
 zend_bool phongo_writeerror_init(zval *return_value, bson_t *bson TSRMLS_DC) /* {{{ */
 {
 	bson_iter_t iter;
-	php_phongo_writeerror_t *writeerror;
+	php_phongo_writeerror_t *intern;
 
-	writeerror = Z_WRITEERROR_OBJ_P(return_value);
+	object_init_ex(return_value, php_phongo_writeerror_ce);
+
+	intern = Z_WRITEERROR_OBJ_P(return_value);
 
 	if (bson_iter_init_find(&iter, bson, "code") && BSON_ITER_HOLDS_INT32(&iter)) {
-		writeerror->code = bson_iter_int32(&iter);
+		intern->code = bson_iter_int32(&iter);
 	}
+
 	if (bson_iter_init_find(&iter, bson, "errmsg") && BSON_ITER_HOLDS_UTF8(&iter)) {
 		uint32_t errmsg_len;
 		const char *err_msg = bson_iter_utf8(&iter, &errmsg_len);
 
-		writeerror->message = estrndup(err_msg, errmsg_len);
+		intern->message = estrndup(err_msg, errmsg_len);
 	}
-	if (bson_iter_init_find(&iter, bson, "errInfo")) {
-		bson_t                 info;
 
-		bson_init(&info);
-		bson_append_iter(&info, NULL, 0, &iter);
+	if (bson_iter_init_find(&iter, bson, "errInfo") && BSON_ITER_HOLDS_DOCUMENT(&iter)) {
+		uint32_t       len;
+		const uint8_t *data = NULL;
+
+		bson_iter_document(&iter, &len, &data);
```

Comment:
`BSON_ITER_HOLDS_DOCUMENT()` and `bson_iter_document()` is consistent with what we were already doing in `phongo_writeconcernerror_init()` for a very similar routine. This replaces the previous `bson_append_iter()` code path, which is much less common.